modules/core/perf/perf_allocation.cpp

```diff
@@ -0,0 +1,44 @@
+#include <array>
+#include "perf_precomp.hpp"
+
+using namespace perf;
+
+#define ALLOC_MAT_SIZES ::perf::szSmall24, ::perf::szSmall32, ::perf::szSmall64, \
+    ::perf::sz5MP, ::perf::sz2K, ::perf::szSmall128, ::perf::szODD, ::perf::szQVGA, \
+    ::perf::szVGA, ::perf::szSVGA, ::perf::sz720p, ::perf::sz1080p, ::perf::sz2160p, \
+    ::perf::sz4320p, ::perf::sz3MP, ::perf::szXGA, ::perf::szSXGA, ::perf::szWQHD, \
+    ::perf::sznHD, ::perf::szqHD
+
+namespace opencv_test
+{
+
+typedef perf::TestBaseWithParam<MatType> MatDepth_tb;
+
+PERF_TEST_P(MatDepth_tb, Allocation_Aligned,
+    testing::Values(CV_8UC1, CV_16SC1, CV_8UC3, CV_8UC4))
```

Comment:
I am not aware of a way to "detect multiple threads during initialization". Because what does "during initialization" mean? Initialization is done by the C++ runtime, not user code. So the user code is not aware of where it is in the overall progress of initialization.

I'm uncomfortable in removing the perf test. Here's my thinking:

1. A custom memory allocator like fastMalloc() is a bedrock critical function...a core feature on top of which other code components are used. I see it used 19 times in the OpenCV 4.5 codebase. And it it used in themost critical places like `class Allocator`, `class Mat`, and `class UMat`. Therefore, I believe it important to always run performance tests on core components and do it isolated.
2. This specific performance test helps evaluate forthcoming updates. I have pending the native Windows aligned memory PR. I will need this exact same test in the code. And I need to run the same test cases yet this time to alter the envvar `OPENCV_ENABLE_MEMALIGN`.
3. Performance testing is not a one-time thing. Performance testing is an ongoing process https://www.google.com/search?q=performance+testing+is+an+ongoing+process

Reply:
Perhaps leave the perf test cpp, and disable it. Then the dev/testers for the Windows PR, or a fix for the GNULIBC issue is made, they can re-enable it for their testing.

Reply:
> and disable it

Agreed.